Framework/MDAlgorithms/src/ConvertToDiffractionMDWorkspace2.cpp

```diff
@@ -247,5 +254,39 @@ void ConvertToDiffractionMDWorkspace2::exec() {
   this->setProperty("OutputWorkspace", iOut);
 }
 
+/** Calculate the extents to use for the conversion from the input workspace.
+ *
+ * @param minVal :: the minimum bounds of the MD space.
+ * @param maxVal :: the maximum bounds of the MD space.
+ */
+void ConvertToDiffractionMDWorkspace2::calculateExtentsFromData(
+    std::vector<double> &minVal, std::vector<double> &maxVal) {
+  auto alg = createChildAlgorithm("ConvertToMDMinMaxLocal");
+  alg->initialize();
+  alg->setProperty<MatrixWorkspace_sptr>("InputWorkspace",
+                                         getProperty("InputWorkspace"));
+  alg->setPropertyValue("QDimensions", "Q3D");
+  std::vector<std::string> dE_modes = Kernel::DeltaEMode::availableTypes();
+  alg->setPropertyValue("dEAnalysisMode",
+                        dE_modes[Kernel::DeltaEMode::Elastic]);
+
+  std::string TargetFrame, Scaling;
+  convertFramePropertyNames(getPropertyValue("OutputDimensions"), TargetFrame,
+                            Scaling);
+  alg->setProperty("Q3DFrames", TargetFrame);
+  alg->setProperty("QConversionScales", Scaling);
+
+  alg->execute();
```

Comment:
How long does this typically take to run? This sub-algorithm is not reporting progress I think.